### ISSUE TITLE
[Linux/ARM] Access permission issue for System.Diagnostics.Process.GetProcessesByName()/GetProcesses()

### ISSUE THREAD

**Issue by JongHeonChoi** [dotnet/runtime#47566]
### Description
I ran the test code in an environment with smack applied(Tizen, 32-bit).
I called two method(```System.Diagnostics.Process.GetProcessesByName()```, ```System.Diagnostics.Process.GetProcesses()```).

* Sample code:
```
static void Main(string[] args)
{
    string name = System.Diagnostics.Process.GetCurrentProcess().ProcessName;
    Console.WriteLine(System.Diagnostics.Process.GetProcessesByName(name));
}
```

It seems to be a problem that occurs when the test app accesses the processes under ```/proc/```.
The following exception occurred. 
* Exception:
```
Unhandled exception. System.UnauthorizedAccessException: Access to the path '/proc/1/stat' is denied.
 ---> System.IO.IOException: Permission denied
   --- End of inner exception stack trace ---
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream.OpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, Boolean useAsync)
   at Interop.procfs.TryReadFile(String filePath, ReusableTextReader reusableReader, String& fileContents)
   at Interop.procfs.TryParseStatFile(String statFilePath, ParsedStat& result, ReusableTextReader reusableReader)
   at System.Diagnostics.Process.GetProcessesByName(String processName, String machineName)
   at System.Diagnostics.Process.GetProcessesByName(String processName)
   at ProcessTest.Program.Main(String[] args)
```

* Analysis:
```at Interop.procfs.TryReadFile(String filePath, ReusableTextReader reusableReader, String& fileContents)```
When I checked the part where the exception occurred, only ```IOException``` is handling the exception.
However, ```UnauthorizedAccessException``` is occurring when calling the ```FileStrem``` method.
Also, the document of the associated API is not mentioned.

So I think not only ```IOException``` but also ```UnauthorizedAccessException``` should be added.

**Comment by adamsitnik** [comment]
We agree that it's a bug. We should most probably replace the following catch block:

https://github.com/dotnet/runtime/blob/a227f0053ee85c651a4c1a37075369eee5800bd6/src/libraries/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.TryReadStatusFile.cs#L184-L188

with sth like

```cs
catch (Exception ex) when (ex is IOException || ex.InnerException is IOException)
{
   contents = null;
   return false;
}
```

@JongHeonChoi would you be interested in sending a PR?

**Comment by JongHeonChoi** [comment]
> @JongHeonChoi would you be interested in sending a PR?

Yes, I will upload the modified PR.
Thanks :)

**Comment by JongHeonChoi** [comment]
@adamsitnik 
Bug Fixed PR : #47566
